### PR TITLE
Change Con.mark to marks

### PR DESCRIPTION
**Pull request by xenomachina**
In newer versions of i3 containers can have multiple marks, so the property
name has changed from mark to marks.

### CODE REVIEW

**Comment by acrisci** [comment]
This is technically a breaking change so I'll have to bump the major version. See [i3#2014](https://github.com/i3/i3/issues/2014)

**Comment by xenomachina** [comment]
Yeah, I was wondering if there might be some way to handle the changeover more gracefully, but I was not able to think of any other possibilities that didn't also have surprising behavior.

**Comment by acrisci** [comment]
I don't see how it can be helped. This is better imo than going our own way with the attribute names or duplicating the property but I hate to break peoples code. Luckily, I don't think marks are a very widely used feature of i3.

I wish I had been paying attention to i3 core at this time to protest the breaking of the ipc interface.

File: i3ipc.py [hunk 1]
```diff
@@ -413,6 +413,15 @@ def __init__(self, data, parent, conn):
             else:
                 setattr(self, attr, None)
 
+        # Make sure marks is never None (use []) to simplify client code. Older
+        # versions of i3 used mark instead of marks, so collapse that into
+        # marks to provide a consistent API.
+        if not self.marks:
```
Comment (acrisci) [inline i3ipc.py]: `self.mark` (the original) is set to `None` by the above code and not deleted in the new version of the ipc. The Con should rather not have the property at all in this case.

Comment (xenomachina) [inline i3ipc.py]: My thinking was that this would let people use the new i3ipc-python with an old version of i3, so they wouldn't have to upgrade the two in lockstep. With old versions of i3 the marks attribute will contain the 0 or 1 marks from i3's mark attribute.

If you prefer, I'm happy to remove mark entirely -- it would certainly simplify the code somewhat, and it probably isn't common that people want to update i3ipc-python ahead of i3.

Comment (xenomachina) [inline i3ipc.py]: By the way, I kind of messed up and included examples/avoid-ws.py in this
pull request. I'm kind of new to pull-requests to be honest, so I didn't
realize that other changes in the same branch automatically got included.
Let me know if you'd like me to remove it.

Comment (acrisci) [inline i3ipc.py]: No, I'd rather make this consistent with the new api across any i3 version since it's a very easy translation.

The inconsistency happens like this:

new api:

```python
con.mark # None
con.marks # [ list of marks... ]
```

old api:

```python
con.mark # attribute error
con.marks # singleton list
```

`con.mark` should not be a property of the Con and it should always throw an attribute error.

Comment (acrisci) [inline i3ipc.py]: Yes, that would be good if you removed it. I like [git undo](https://github.com/tj/git-extras/blob/master/Commands.md#git-undo) from git-extras for that.

Also, you should be doing pull requests in a topic branch not the master branch.

Comment (xenomachina) [inline i3ipc.py]: Thanks for the git-undo tip. That's nice.

Yes, I realized I should be doing pull requests in a topic branch once I
discovered that they just slurp in everything from the branch they were
created on. :-)

Do you want me to re-do this pull request on a topic branch, or is there
some way to change the branch that a pull request is tracking, or are you
okay with this one being on master?

Comment (acrisci) [inline i3ipc.py]: Now that we are here, it's fine. But the script would be better in another pull request since it's a different issue that should have its own discussion.

**Comment by xenomachina** [comment]
Ok, I've removed that other commit from this branch, and then removed handling of the old "mark" attribute from Con.

**Comment by acrisci** [comment]
Thanks. I'll make a decision on how to handle the old api and make that change before the 2.0 release :+1:
